src/Native/System.Native/pal_networking.h

```diff
@@ -409,7 +409,7 @@ extern "C" Error CreateSocketEventBuffer(int32_t count, SocketEvent** buffer);
 
 extern "C" Error FreeSocketEventBuffer(SocketEvent* buffer);
 
-extern "C" Error TryChangeSocketEventRegistration(int32_t port, int32_t socket, int32_t currentEvents, int32_t newEvents, uintptr_t data);
+extern "C" Error TryChangeSocketEventRegistration(int32_t port, int32_t socket, int32_t currentEvents, int32_t newEvents, int64_t data);
```

Comment:
I wasn't able to test on Linux before. Now I have the access. On FreeBSD and OSX it was compiling.
I will add a #if #else guard, without changing the function signature.